tableau/v_tableau.zendesk_tickets.sql

```diff
@@ -1,39 +1,135 @@
+  )
 
 SELECT t.id AS ticket_id
       ,CONVERT(VARCHAR(500), t.[subject]) AS ticket_subject
-      ,t.[status] AS ticket_status
-      ,t.custom_location AS [location]
       ,t.created_at
-      ,t.updated_at
-      ,t.due_at
+      ,t.[status] AS ticket_status
       ,t.custom_category AS category
       ,t.custom_tech_tier AS tech_tier
-      ,t.group_id
-      ,t.submitter_id
-      ,t.assignee_id
+      ,t.custom_location AS [location]
+      ,CONCAT('https://teamschools.zendesk.com/agent/tickets/',t.id) AS ticket_url
+      ,DATEDIFF(WEEKDAY,t.created_at,gu.group_updated) AS weekdays_created_to_last_group
+      ,DATEDIFF(WEEKDAY,t.created_at,tm.solved_at) AS weekdays_created_to_solved
 
       ,s.[name] AS submitter_name
 
-      ,a.[name] AS assignee_name
-      ,a.custom_user_group AS assignee_user_group
+      ,a.[name] AS assignee
 
-      ,g.[name] AS group_name
+      ,c.primary_job AS assignee_primary_job
+      ,c.primary_site AS assignee_primary_site
+      ,c.legal_entity_name AS assignee_legal_entity
 
+      ,tm.assignee_updated_at
+      ,tm.initially_assigned_at
+      ,tm.solved_at
       ,tm.replies AS comments_count
-      ,tm.solved_at AS solved_timestamp
       ,tm.full_resolution_time_in_minutes_business AS total_bh_minutes
       ,tm.reply_time_in_minutes_business
+      ,DATEDIFF(WEEKDAY,t.created_at, tm.initially_assigned_at) AS weekdays_created_to_first_assigned
+      ,DATEDIFF(WEEKDAY,t.created_at,tm.assignee_updated_at) AS weekdays_created_to_last_assigned
+
+      ,gu.group_updated AS group_updated
+
+      ,oa.assignee_name AS original_assignee
+      ,oa.orig_assignee_job
+
+      ,sx.primary_on_site_department AS submitter_dept
+      ,sx.primary_job AS submitter_job
+      ,sx.primary_site AS submitter_site
+      ,sx.legal_entity_name AS submitter_entity
+
+      ,CASE
+        WHEN tm.assignee_stations < tm.group_stations THEN og.group_name
+        WHEN oa.orig_assignee_dept IS NULL THEN og.group_name
+        ELSE oa.orig_assignee_dept END AS og_group
+        --if fewer assignees than groups, then the original Zendesk group, if original assignee's ADP department null then the Zendesk group, otherwise, the original assignee's ADP department
+       ,CASE 
+        WHEN tm.assignee_stations < tm.group_stations THEN g.[name]
+        WHEN c.primary_on_site_department IS NULL THEN g.[name]
+        ELSE c.primary_on_site_department END AS last_assigned_dept_group -- if fewer assignees than groups, then the Zendesk group, if ADP department is null then the Zendesk group, otherwise, the last assignee's ADP department
+
 FROM gabby.zendesk.ticket t
 LEFT JOIN gabby.zendesk.[user] s
   ON t.submitter_id = s.id
 LEFT JOIN gabby.zendesk.[user] a
   ON t.assignee_id = a.id
 LEFT JOIN gabby.zendesk.[group] g
   ON t.group_id = g.id
+LEFT JOIN gabby.people.staff_crosswalk_static c
+  ON a.email = c.userprincipalname
+--LEFT JOIN gabby.people.staff_crosswalk_static sx
+--  ON s.email = c.userprincipalname
```

Comment:
your `ON` clause points to the wrong table alias (`c` instead of `sx`). That's prob why you're getting dupes.

Reply:
gah. fixed.